File: core-platform/apps/core-webapp/src/services/epic.service.ts
import axiosInstance from "../axiosInstance";
import type { Epic, EpicDTO } from "../types/epic.types";

const EPIC_API_BASE = "/epic";

export const epicService = {
  createEpic: async (data: EpicDTO): Promise<Epic> => {
    const response = await axiosInstance.post(EPIC_API_BASE, data);
    return response.data;
  },

  updateEpic: async (id: number, data: EpicDTO): Promise<Epic> => {
    const response = await axiosInstance.put(`${EPIC_API_BASE}/${id}`, data);
    return response.data;
  },

  deleteEpic: async (id: number): Promise<void> => {
    await axiosInstance.delete(`${EPIC_API_BASE}/${id}`);
  },

  getEpicById: async (id: number): Promise<Epic> => {
    const response = await axiosInstance.get(`${EPIC_API_BASE}/${id}`);
    return response.data;
  },

  getAllEpics: async (organizationId: number): Promise<Epic[]> => {
    const response = await axiosInstance.get(EPIC_API_BASE, {
      params: { organizationId },
    });
    return response.data;
  },

  getEpicsByProject: async (projectId: number): Promise<Epic[]> => {
    const response = await axiosInstance.get(
      `${EPIC_API_BASE}/project/${projectId}`,
    );
    return response.data;
  },

  getEpicByKey: async (key: string): Promise<Epic> => {
    const response = await axiosInstance.get(`${EPIC_API_BASE}/key/${key}`);
    return response.data;
  },
};

File: core-platform/services/core-service/src/main/java/com/dev/core/service/EpicService.java
package com.dev.core.service;

import com.dev.core.model.EpicDTO;

import java.util.List;

public interface EpicService {
    EpicDTO createEpic(EpicDTO dto);
    EpicDTO updateEpic(Long id, EpicDTO dto);
    void deleteEpic(Long id);
    EpicDTO getEpicById(Long id);
    List<EpicDTO> getAllEpics(Long organizationId);
    List<EpicDTO> getEpicsByProject(Long projectId);
    EpicDTO getEpicByKey(String key);
}
